src/ctap2_capi.rs

```diff
@@ -909,3 +922,104 @@ pub unsafe extern "C" fn rust_ctap2_sign_result_username_copy(
         Err(_) => false,
     }
 }
+
+/// # Safety
+///
+/// This function is used to get how long the JSON-representation of a status update is.
+#[no_mangle]
+pub unsafe extern "C" fn rust_ctap2_status_update_len(
+    res: *const StatusUpdate,
+    len: *mut size_t,
+) -> bool {
+    if res.is_null() || len.is_null() {
+        return false;
+    }
+
+    match serde_json::to_string(&*res) {
+        Ok(s) => {
+            *len = s.len();
+            true
+        }
+        Err(e) => {
+            error!("Failed to parse {:?} into json: {:?}", &*res, e);
+            false
+        }
+    }
+}
+
+/// # Safety
+///
+/// This method does not ensure anything about dst before copying, so
+/// ensure it is long enough (using rust_ctap2_status_update_len)
+#[no_mangle]
```

Comment:
I think I'd prefer an API where you always take a length, check if it would fit, and if not return (via out-param) the required length. It still requires two separate calls most of the time, but I think it's safer to treat the pointer+len as two inseparable things.

Reply:
The 'problem' here is that all other C-API functions currently behave like this. Should I rewrite them all, or just the new ones?
Since the new ones for status updates are now using JSON and I could experiment with that on the C++ side, I'm more and more inclined to throw away most of the API and just return JSON all the time :) But I would need to try this out first (e.g. serializing byte arrays would need to use base64)

Reply:
Ok, I have only changed the new function your proposed interface. If I should do the others as well, just reopen this discussion.

Reply:
You're right, this creates an inconsistency with the rest. It's not that big a deal anyway, let's just leave it all in the 2-function style, then? Sorry about going back and forth on this :)

Reply:
I already changed it for the new function :-D (see [b4342e9](https://github.com/mozilla/authenticator-rs/pull/163/commits/b4342e90b6c09a370ebdc7a8e7638f6a2d9e5463)) Rollback or leave it? (I don't care either way)

Reply:
Done